packages/bitcore-node/test/integration/models/block.integration.ts

```diff
@@ -0,0 +1,611 @@
+      expect(ownBlock.version).to.equal(536870912);
+      // TODO: assertion for block times
+      expect(ownBlock.transactionCount).to.equal(1);
+      expect(ownBlock.processed).to.equal(true);
+
+      logger.info(`new block was successfully added with hash`, ownBlock.hash);
+
+      const transaction = await TransactionModel.find({
+        chain: 'BTC',
+        network: 'regtest',
+        blockHash: '64bfb3eda276ae4ae5b64d9e36c9c0b629bc767fb7ae66f9d55d2c5c8103a929'
+      }).exec();
+      expect(transaction.length).to.equal(1);
+      expect(transaction[0].chain).to.equal('BTC');
+      expect(transaction[0].network).to.equal('regtest');
+      expect(transaction[0].txid).to.equal('08e23107e8449f02568d37d37aa76e840e55bbb5f100ed8ad257af303db88c08');
+      expect(transaction[0].blockHash).to.equal('64bfb3eda276ae4ae5b64d9e36c9c0b629bc767fb7ae66f9d55d2c5c8103a929');
+      expect(transaction[0].blockHeight).to.equal(9);
+      expect(transaction[0].coinbase).to.equal(true);
+      expect(transaction[0].locktime).to.equal(0);
+      expect(transaction[0].size).to.equal(0);
+      // TODO: assertion for block times
+      expect(transaction[0].wallets).to.be.an('array').that.is.empty;
+
+      logger.info(`tx: ${transaction[0].txid} was successfully stored in the TX model`);
+
+    });
+  });
+
+  describe('getLocalTip', () => {
+    it('should get the highest processed block for a particular chain and network', async () => {
+
+      await BlockModel.create({ processed: true, chain: 'BTC', network: 'regtest', height: 1 });
+      await BlockModel.create({ processed: true, chain: 'BTC', network: 'regtest', height: 2 });
+      await BlockModel.create({ processed: true, chain: 'BTC', network: 'regtest', height: 3 });
+
+      const result = await BlockModel.getLocalTip({ chain: 'BTC', network: 'regtest' });
+
+      expect(result.height).to.equal(3);
+      expect(result.processed).to.equal(true);
+      expect(result.chain).to.equal('BTC');
+      expect(result.network).to.equal('regtest');
+
+    });
+    it('should return block with height zero if block is not found for a particular chain and network', async () => {
+
+      await BlockModel.create({ processed: true, chain: 'BTC', network: 'regtest', height: 1 });
+      await BlockModel.create({ processed: true, chain: 'BTC', network: 'regtest', height: 2 });
+      await BlockModel.create({ processed: true, chain: 'BTC', network: 'regtest', height: 3 });
+      await BlockModel.create({ processed: true, chain: 'BCH', network: 'regtest', height: 10 });
+      await BlockModel.create({ processed: true, chain: 'BTC', network: 'testnet', height: 10 });
+
+      const result = await BlockModel.getLocalTip({ chain: 'ETH', network: 'regtest' });
+
+      expect(result.height).to.equal(0);
+
+    });
+  });
+
+  describe('getLocatorHashes', () => {
+    it('should return the highest processed block hash for the last thirty blocks for a particular chain and network', async () => {
```

Comment:
If you test title says that this should be for the last 30 blocks then you need to insert 30 blocks. For 30 it's ok to use a simple loop.  It's also ok if you don't have a real hash for this test and just do `hash1` `hash2` and so on.

```
it('should return the highest....', async(){

for(let i = 1; i < 31; i++){
await BlockModel.create({processed: true, chain: 'BTC', network: 'regtest', height: i , hash: `hash${i}`);
}

// then your test and expects

}

```

Reply:
resolved

Reply:
confirmed